### PR TITLE
Fix profiles datagrid

### PR DESCRIPTION
**Pull request by dreadnip**
## Type

- Non critical bugfix

## Resolves the following issues

fixes https://github.com/forkcms/forkcms/issues/3100

## Pull request description

Fixes a couple of inconsistencies when handling profiles/groups from the mass action select.

### CODE REVIEW

File: src/Backend/Modules/Profiles/Actions/Index.php [hunk 1]
```diff
@@ -130,14 +130,26 @@ private function loadDataGrid(): void
         );
         $this->dgProfiles->setColumnFunction('htmlspecialchars', ['[display_name]'], 'display_name', false);
 
+        // check if
+        $groups = BackendProfilesModel::getGroups();
```
Comment (tijsverkoyen) [inline src/Backend/Modules/Profiles/Actions/Index.php:134]: Would't I be more logical to have the $actions populated at the start. So:

```
$actions = [
  'delete' => BL::getLabel('Delete'),
];

if (count($groups) > 0) {
  $actions['addToGroup'] = BL::getLabel('AddToGroup');
}
```

Comment (dreadnip) [reply src/Backend/Modules/Profiles/Actions/Index.php]: Updated in 22b604e

**Review by tijsverkoyen — APPROVED** [review]

File: src/Backend/Modules/Profiles/Engine/Model.php [hunk 2]
```diff
@@ -691,9 +691,15 @@ public static function insertProfileGroup(array $membership): int
         $profile = BackendModel::get('profile.repository.profile')->find($membership['profile_id']);
         $group = BackendModel::get('profile.repository.profile_group')->find($membership['group_id']);
 
+        $startsOn = new DateTime();
+        $startsOn->setTimestamp($membership['starts_on']);
+        $startsOn->format('Y-m-d H:i:s');
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Engine/Model.php:696]: you are not using the output here.

File: src/Backend/Modules/Profiles/Engine/Model.php [hunk 3]
```diff
@@ -691,9 +691,15 @@ public static function insertProfileGroup(array $membership): int
...
+
         $expiresOn = null;
         if (array_key_exists('expires_on', $membership)) {
-            $expiresOn = DateTime::createFromFormat('Y-m-d H:i:s', $membership['expires_on']);
+            $expiresOn = new DateTime();
+            $expiresOn->setTimestamp($membership['expires_on']);
+            $expiresOn->format('Y-m-d H:i:s');
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Engine/Model.php:702]: you are not using the output here.

**Review by carakas — CHANGES_REQUESTED** [review]

File: src/Backend/Modules/Profiles/Engine/Model.php (hunk 2 above)
Comment (dreadnip) [reply src/Backend/Modules/Profiles/Engine/Model.php]: Fixed in 3a36fa8

File: src/Backend/Modules/Profiles/Engine/Model.php (hunk 3 above)
Comment (dreadnip) [reply src/Backend/Modules/Profiles/Engine/Model.php]: Fixed in 3a36fa8

**Comment by dreadnip** [comment]
3a36fa8 should fix the datagrid mass actions add to group behaviour. While testing I noticed that the profile/group add/update flow is broken. I didn't dig deep but it seemed like the form fields still used the old notation `expiration_date`, while the back-end code expected `expiry_on`. Should be looked into.

**Review by carakas — APPROVED** [review]

**Review by carakas — CHANGES_REQUESTED** [review]
@dreadnip there is a failing test for 7.4, can you fix this one please?

**Comment by dreadnip** [comment]
Removed a couple of old strtotime() instances in the test. Should be okay now.

**Review by carakas — APPROVED** [review]

**Pull request merged**